installers/macOS/setup.py

```diff
@@ -0,0 +1,201 @@
+pyls :
+    <path>/Contents/MacOS/python: No module named pyls
+    Note: still occurs in alias mode
+qtawesome :
+    NotADirectoryError: [Errno 20] Not a directory: '<path>/Resourses/lib/
+    python38.zip/qtawesome/fonts/fontawesome4.7-webfont.ttf'
+spyder :
+    NotADirectoryError: [Errno 20] Not a directory: '<path>/Resources/lib/
+    python38.zip/spyder/app/mac_stylesheet.qss'
+spyder_kernels :
+    No module named spyder_kernels.console.__main__
+sphinx :
+    No module named 'sphinx.builders.changes'
+
+"""
+
+import os
+import sys
+import shutil
+import argparse
+import pkg_resources
+from logging import getLogger, StreamHandler, Formatter
+from setuptools import setup
+from dmgbuild import build_dmg
+
+# parse additional arguments for setup.py
+parser = argparse.ArgumentParser()
+parser.add_argument('--lite', dest='make_lite', action='store_true',
+                    default=False, help='Build with minimal internal packages')
+parser.add_argument('--dmg', dest='make_dmg', action='store_true',
+                    default=False, help='Create DMG')
+
+# parse additional arguments for py2app
+subparsers = parser.add_subparsers()
+py2app_parser = subparsers.add_parser('py2app')
+py2app_parser.add_argument('--make-app', dest='make_app',
+                           help=argparse.SUPPRESS)
+
+args, _ = parser.parse_known_args()
+[sys.argv.remove(s) for s in ['--lite', '--dmg'] if s in sys.argv]
+
+# setup logger
+fmt = Formatter('%(asctime)s [%(levelname)s] [%(name)s] -> %(message)s')
+h = StreamHandler()
+h.setFormatter(fmt)
+logger = getLogger('spyder-macOS')
+logger.addHandler(h)
+logger.setLevel('INFO')
+
+# setup paths
+here = os.path.abspath(__file__)
+thisdir = os.path.dirname(here)
+spy_repo = os.path.realpath(os.path.join(thisdir, '..', '..'))
+distdir = os.path.join(thisdir, 'dist')
+
+# symlink to spyder package
+spy_link = os.path.join(thisdir, 'spyder')
+if not os.path.islink(spy_link):
+    os.symlink(os.path.join(spy_repo, 'spyder'), spy_link)
+
+# =============================================================================
+# App Creation
+# =============================================================================
+from spyder import __version__ as spy_version                    # noqa
+from spyder.config.utils import EDIT_FILETYPES, _get_extensions  # noqa
+from spyder.config.base import MAC_APP_NAME                      # noqa
+
+iconfile = os.path.join(spy_repo, 'img_src', 'spyder.icns')
+
+py_ver = [sys.version_info.major, sys.version_info.minor,
+          sys.version_info.micro]
+
+PACKAGES = ['alabaster', 'astroid', 'ipykernel', 'IPython', 'jedi', 'jinja2',
+            'keyring', 'parso', 'pygments', 'pyls', 'qtawesome', 'spyder',
+            'spyder_kernels', 'sphinx',
+            ]
+
+if args.make_lite:
+    INCLUDES = []
+    EXCLUDES = ['numpy', 'scipy', 'pandas', 'matplotlib', 'cython', 'sympy']
+else:
+    INCLUDES = ['numpy', 'scipy', 'pandas', 'matplotlib', 'cython', 'sympy']
+    EXCLUDES = []
```

Comment:
I think pip should be excluded in both cases (in case it's included) to forbid people messing with the installer.